modules/govuk_postgresql/manifests/wal_e/restore.pp

```diff
@@ -0,0 +1,133 @@
+# == Define: Govuk_postgresql::Wal_e::Restore
+#
+# Allows restoring from an offsite backup of a PostgreSQL database
+# from an S3 bucket using WAL-E: https://github.com/wal-e/wal-e
+#
+# === Parameters:
+#
+# [*aws_access_key_id*]
+#   The AWS access ID for the user that is allowed to
+#   access the S3 bucket.
+#
+# [*aws_secret_access_key*]
+#   The AWS secret access key for the user that is allowed
+#   to access the S3 bucket.
+#
+# [*s3_bucket_url*]
+#   The unique address of the S3 bucket, in the format of:
+#   s3://bucketaddress/directory/foo
+#
+# [*aws_region*]
+#   The AWS region where the specified bucket resides.
+#   Default: eu-west-1
+#
+# [*wale_private_gpg_key*]
+#   The private GPG to import to allow encryption. If present, WAL-E will
+#   will encrypt the backups.
+#
+# [*wale_private_gpg_key_fingerprint*]
+#   The GPG key fingerprint for the above GPG key.
+#
+# [*db_dir*]
+#   The database directory to backup. WAL-E does hot backups
+#   so there should be no impact, and the default backups up
+#   everything.
+#   Default: /var/lib/postgresql/9.3/main
+#
+define govuk_postgresql::wal_e::restore (
+  $aws_access_key_id,
+  $aws_secret_access_key,
+  $s3_bucket_url,
+  $aws_region = 'eu-west-1',
+  $wale_private_gpg_key = undef,
```

Comment:
I suspect we don't need the `wale` prefix on these two variables since we're already in the `wal_e` class.

Reply:
Same reasoning as above.  I took the current names to be consistent with the classes which exist, and to avoid needing to also change secrets to make everything consistent.

Reply:
Sure, that sounds reasonable!